src/components/Toast/ToastContext.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { createContext, PropsWithChildren, useContext, useMemo, useState } from "react";
+import { Toast } from "./Toast";
+import { ToastNoticeProps } from "./ToastNotice";
+
+export type ToastContextProps = {
+  setNotice: React.Dispatch<React.SetStateAction<ToastNoticeProps>>;
+};
+
+export const ToastContext = createContext<ToastContextProps>({
+  setNotice: () => {},
+});
+
+export function ToastProvider(props: PropsWithChildren<{}>) {
+  const [notice, setNotice] = useState<ToastNoticeProps>({} as ToastNoticeProps);
```

Comment:
Suggestion/question: can we avoid the `as ToastNoticeProps`? I'm worried that we're letting a `notice` that the type system _thinks_ is a `ToastNoticeProps` (but is not actually) into the runtime that later will lead to weirdness/bugs.

Maybe this should be `useState<ToastNoticeProps | undefined>(undefined)` as a better signal for "we don't have a notice right now"?